File: selectionSort.c
#include "selectionSort.h"
void selectionSort(int list[10], int n) {
    int i,j,x,c;

    for(i=0; i<n; i++) {
        x = list[i];
        c = -1;
        j = i+1;
        while(j<n) {
            if(x < list[j]) {
                x = list[j];
                c = j;
            }
            j++;
        }

        if(c != -1) {
            list[c] = list[i];
            list[i] = x;
        }
    }
}

File: selectionSort.java
public class selectionSort extends sortArray{
    static void swap(int[] ar, int i, int j) {
        int t;
        t = ar[i];
        ar[i] = ar[j];
        ar[j] = t;
    }
    public void sort(int[] list) {
        System.out.println("Inside SelectionSort");
        int n = list.length;
        int x,t;

        for(int i=0; i<n-1; i++){
            x = list[i];
            t = i;
            for(int j=i+1; j<n; j++){
                if(list[j] < x) {
                    x = list[j];
                    t = j;
                }
            }
            if(t != i) swap(list, i, t);
        }
        System.out.println("SelectionSort Ended.\n");
    }
}
